### PR TITLE
cli-test-fixup

### PR DESCRIPTION
**Pull request by maddenp-noaa**
Fixes per group discussion today. See inline comments.

### CODE REVIEW

File: src/uwtools/cli.py [hunk 1]
```diff
@@ -42,17 +42,17 @@ def main() -> None:
     setup_logging(quiet=True)
     try:
         args, checks = _parse_args(sys.argv[1:])
-        for check in checks[args.mode][args.submode]:
+        for check in checks[vars(args)[STR.mode]][vars(args)[STR.submode]]:
```
Comment (maddenp-noaa) [inline src/uwtools/cli.py:45]: Replace references to `args` members where the expression uses the equivalent of an uncheckable string literal with a reference to the `STR` object, which can be verified by the linter. Many repetitions of this pattern follow.

File: src/uwtools/cli.py [hunk 2]
```diff
@@ -479,7 +481,7 @@ def _add_arg_config_file(group: Group) -> None:
 
 def _add_arg_cycle(group: Group) -> None:
     group.add_argument(
-        "--cycle",
+        _switch(STR.cycle),
```
Comment (maddenp-noaa) [inline src/uwtools/cli.py:484]: A string literal slipped in here -- relocated to `STR`.

File: src/uwtools/cli.py [hunk 3]
```diff
@@ -298,12 +300,12 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
 
     :param args: Parsed command-line args.
     """
-    forecast_class = uwtools.drivers.forecast.CLASSES[args.forecast_model]
+    forecast_class = uwtools.drivers.forecast.CLASSES[vars(args)[STR.model]]
```
Comment (maddenp-noaa) [inline src/uwtools/cli.py:303]: I think `forecast_model` was replaced with `model` somewhere along the line but this didn't get updated. Fix, and fixed corresponding unit test.

File: src/uwtools/tests/test_cli.py [hunk 4]
```diff
@@ -195,15 +195,20 @@ def test__dispatch_config(params):
...
     vars(args).update({STR.file1path: 1, STR.file1fmt: 2, STR.file2path: 3, STR.file2fmt: 4})
     with patch.object(cli.uwtools.config.core, "compare_configs") as compare_configs:
         cli._dispatch_config_compare(args)
-    assert compare_configs.called_once_with(args)
+    compare_configs.assert_called_once_with(
+        config_a_path=args.file_1_path,
+        config_a_format=args.file_1_format,
+        config_b_path=args.file_2_path,
+        config_b_format=args.file_2_format,
+    )
```
Comment (maddenp-noaa) [inline src/uwtools/tests/test_cli.py:211]: Fixed several instances of improper assertions:
1. Replaced `assert` statements with invocations of mock objects' `asssert_called_once_with()` methods.
2. With the previous change in place, tests (correctly) started failing, so updated expected call arguments with the correct values.

This pattern repeats throughout the remainder of this module.

**Review by WeirAE — APPROVED** [review]
Whew, there were even more than expected, I'm glad we took the time to look through everything - LGTM

File: src/uwtools/cli.py (hunk 2 above)
Comment (christinaholtNOAA) [reply src/uwtools/cli.py:484]: 😬 My bad!

**Review by christinaholtNOAA — APPROVED** [review]
Thanks for the quick turnaround on the tedious work. Looks good.

**Review by christinaholtNOAA — APPROVED** [review]
Even better!

**Pull request merged**